ntbs-service/Services/AlertService.cs

```diff
@@ -118,22 +118,39 @@ public async Task DismissAllOpenAlertsForNotification(int notificationId)
 
             foreach (var alert in alerts)
             {
+                // notification may have a duplicate alert to be dismissed on multiple other records
                 if (alert is DataQualityPotentialDuplicateAlert duplicateAlert)
                 {
-                    var duplicate =
-                        await _notificationRepository.GetNotificationForAlertCreationAsync(duplicateAlert.DuplicateId);
+                    await DismissDuplicationAlertsOnDuplicateRecord(notificationId, duplicateAlert.DuplicateId);
+                }
+                
+                CloseAlert(alert);
+            }
 
-                    if (duplicate.NotificationStatus.IsOpen())
+            await _alertRepository.SaveAlertChangesAsync();
+        }
+
+        private async Task DismissDuplicationAlertsOnDuplicateRecord(int notificationId, int duplicateAlertId)
+        {
+            var duplicateRecordAlerts = await _alertRepository.GetAllOpenAlertsByNotificationId(duplicateAlertId);
+            if (duplicateRecordAlerts != null)
```

Comment:
Would be a little more readable with `.Any()`

Reply:
I tried this, and unfortunately calling .Any() on a null object (as opposed to a list with no elements) gives an ArgumentNullException. I've changed it to `!(duplicateRecordAlerts is null)`. `duplicateRecordAlerts is not null` would be best but we need language version 9.0 for that to work

Reply:
Ah my mistake, you're looking for: `duplicateRecordAlerts?.Any()`